src/main.cpp

```diff
@@ -430,15 +416,38 @@ static std::tuple<string, string> _create_cert(const Options &options, const str
 		csr = _spawn("openssl req -new -config " + cnf + " -key " + pkey_file);
 		return std::make_tuple(pkey, csr);
 	} else {
+		// Initialize the HSM and create a key in it.
+		_setenv("OPENSSL_CONF", cnf.c_str());
+
+		/* Just in case we are using a TPM */
+		_setenv("TPM2_PKCS11_STORE", "/var/tpm2_pkcs11");
+
+		/* Remove the database, we dont care if it fails */
+		g_spawn_command_line_sync("rm /var/tpm2_pkcs11/tpm2_pkcs11.sqlite3",
```

Comment:
One issue here is that we're deleting the tpm2 pkcs11 database even when not used (e.g. a user registering without using tpm2 will have its pkcs11 tpm2 database erased as consequence, which is not ideal).

Reply:
yes, we probably need an extra option for TPM.

btw I also need to update https://github.com/foundriesio/lmp-device-register/blob/02b6d6663135c9e4d8cfa0c5f9a23192444db778/src/main.cpp#L729